src/main/java/jsr223/perl/PerlScriptEngineFactory.java

```diff
@@ -36,22 +36,21 @@
 
 
 public class PerlScriptEngineFactory implements ScriptEngineFactory {
+    private final Map<String, Object> parameters = new HashMap<>();
+
+    private static PerlProcessBuilderUtilities processBuilderUtilities = new PerlProcessBuilderUtilities();
+
+    private static PerlVersionGetter perlVersionGetter = new PerlVersionGetter(processBuilderUtilities);
 
     // Script engine parameters
     private static final String NAME = "perl";
 
     private static final String ENGINE = NAME;
 
-    private static final String ENGINE_VERSION = "0.1.0";
+    private static final String ENGINE_VERSION = perlVersionGetter.getPerlVersion(PerlSingletonPerlProcessBuilderFactory.getInstance());
```

Comment:
could you handle the case where perl is not found ? In that case simply return a default version number maybe.